qutebrowser/browser/urlmarks.py

```diff
@@ -244,17 +264,35 @@ class BookmarkManager(UrlMarkManager):
         - removed gets emitted with the URL as argument.
     """
 
+    def _bookmarks_directory(self):
+        """Return the directory containing the bookmarks.
+
+        For bookmarks are data, return by default a folder inside the standard
+        data directory. In order not to break the compatibility with an older
+        behavior, if bookmarks are found in the standard configuration
+        directory, return this directory instead.
+        """
+        config_dir = standarddir.config()
+        if config_dir is not None:
+            inconfigdir = os.path.join(standarddir.config(), 'bookmarks')
+        else:
+            inconfigdir = None
+        indatadir = os.path.join(standarddir.data(), 'bookmarks')
+        if inconfigdir is not None and os.path.isdir(inconfigdir):
+            return inconfigdir
+        else:
+            return indatadir
```

Comment:
Oh, that's even worse than I thought. Returning early from the parent constructor does not ensure that the object is in failed state or which features are disabled. As long as the object lives, it should be possible to call any of its methods - here `os.path.join` would raise `TypeError` if it gets `None` in the first argument. The check for `None` should be here and I think `_bookmarks_directory` should also just return `None` in that case.